src/main/java/jcmp/DocCompare.java

```diff
@@ -618,11 +620,16 @@ private boolean isIndex(String s) {
     private static class Pair {
         private final int i1,i2;
 
+        @Override
         public int hashCode() {
             return i1*1001+i2;
         }
 
+        @Override
         public boolean equals(Object o) {
+            if (!(o instanceof Pair)) {
```

Comment:
There is cost with creating a stack trace.  It will be faster to not rely on try-catch.